pywbem_mock/_mockmofwbemconnection.py

```diff
@@ -112,38 +102,34 @@ def _setns(self, value):
     )
 
     def EnumerateInstanceNames(self, *args, **kwargs):
-        """This method is used by the MOF compiler only when it creates a
-        namespace in the course of handling CIM_ERR_NAMESPACE_NOT_FOUND.
-        Because the operations of this class silently create every namespace
-        that is needed and never return that error, this method is never
-        called, and is therefore not implemented.
+        """
+        Not Implemented because not used with the MOF compiler.
         """
 
         raise CIMError(
-            CIM_ERR_FAILED, 'This should not happen!',
+            CIM_ERR_FAILED, 'EnumerateInstanceNames not implemented!',
             conn_id=self.conn_id)
```

Comment:
If this class is only used by the mof compiler, and if the mof compiler does not use this method, this is a situation where an assertion error would be appropriate, instead.

Applies to multiple such methods.

Reply:
Agreed.

Reply:
Changed to asserts.  DONE

Reply:
Note that we could also af used the NotImplementedError but this clarifies that the error is from the child class, not from the base class

Reply:
Better Question: Why include the unimplemnted classes in BaseRepositoryConnection since it is just for compiler support?

**DISCUSSION**

Reply:
Ignore this one.

Reply:
We decided to ignore this one. DONE